play-services-fitness/src/main/aidl/com/google/android/gms/fitness/internal/IGoogleFitHistoryApi.aidl

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-FileCopyrightText: 2023 microG Project Team
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.google.android.gms.fitness.internal;
+
+import com.google.android.gms.fitness.request.DataDeleteRequest;
+import com.google.android.gms.fitness.request.GetSyncInfoRequest;
+import com.google.android.gms.fitness.request.DataInsertRequest;
+import com.google.android.gms.fitness.request.DataReadRequest;
+
+interface IGoogleFitHistoryApi {
+    void getDeleteData(in DataDeleteRequest dataDeleteRequest) = 0;
+    void getSyncInfo(in GetSyncInfoRequest getSyncInfoRequest) = 1;
+    void getInsertData(in DataInsertRequest dataInsertRequest) = 2;
+    void getReadData(in DataReadRequest dataReadRequest) = 3;
```

Comment:
Should be:
```aidl
    void readData(in DataReadRequest request) = 0;
    void insertData(in DataInsertRequest request) = 1;
    void deleteData(in DataDeleteRequest request) = 2;
    void getSyncInfo(in GetSyncInfoRequest request) = 3;
```
The functions suffix id is not just there for fun. It's the AIDL method id. If it mismatches (as in this case), the AIDL file doesn't add any value (because the invoked method does not match what is needed). While it may be able to run if the method signature is sufficiently similar as in this case (all of these take a SafeParcelable which - as the name suggests - is safe), a mismatch can also result in crashes.

Beyond those that you already added, the following are already defined as well:
```aidl
    void readStats(in ReadStatsRequest request) = 4;
    void readRaw(in ReadRawRequest request) = 5;
    void getDailyTotal(in DailyTotalRequest request) = 6;
    void insertDataPrivileged(in DataInsertRequest request) = 7;
    void updateData(in DataUpdateRequest request) = 8;
    void registerDataUpdateListener(in DataUpdateListenerRegistrationRequest request) = 9;
    void unregisterDataUpdateListener(in DataUpdateListenerUnregistrationRequest request) = 10;
    void getFileUri(in GetFileUriRequest request) = 11;
    void getDebugInfo(in DebugInfoRequest request) = 12;
```
Please add them as well. If you don't want to do the effort just now to add all the parcelables for this, you can either create empty safe parcelables (all of them are safe parcelables) or just comment out those lines. We should still make sure to list them here in case they're needed in the future and to avoid confusion.

Reply:
To avoid duplicate efforts I will link it since I have already done it for myself (with the commented lines): https://github.com/micro5k/GmsCoreMod/commit/ad44e868a2fb64e1ed734d6bfc03def1b67ad36a.patch

Reply:
Thanks for the reply.  I added the missing methods and tested, verifying that the <Fit> app can be entered normally.